pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/plugincheck/StartupPluginChecker.kt

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023
+
+package com.amos.pitmutationmate.pitmutationmate.plugincheck
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiManager
+import org.apache.commons.io.IOUtils
+import org.codehaus.groovy.ast.builder.AstBuilder
+import java.io.File
+import java.io.FileInputStream
+
+class StartupPluginChecker : ProjectActivity {
+
+    override suspend fun execute(project: Project) {
+        ApplicationManager.getApplication().invokeLater {
+            checkGroovyBuildFile(project)
+            checkKotlinBuildFile(project)
+        }
+    }
+
+    private fun checkKotlinBuildFile(project: Project) {
+        val buildFileName = "build.gradle.kts"
+        val kotlinBuildFile = File(project.basePath + "/$buildFileName")
+        if (kotlinBuildFile.exists()) {
+            val virtualFile = LocalFileSystem.getInstance().findFileByIoFile(kotlinBuildFile)
+            if (virtualFile != null) {
+                val psiFile = PsiManager.getInstance(project).findFile(virtualFile)
+                val pluginCheckerKotlin = PluginCheckerKotlin()
+                psiFile?.node?.psi?.accept(pluginCheckerKotlin)
+                val pitestPluginText = "id(\"info.solidsoft.pitest\") version \"1.15.0\""
+                val companionPluginText = "id(\"io.github.amosproj.pitmutationmate.override\") version \"1.0\""
```

Comment:
It was mentioned like this in the ticket #103.

## Acceptance criteria

- Check if companion plugin is in the project
- Check if Pitest gradle plugin is in the project
- Throw a visible and explaining error message & instruction on how to fix the issue.

Reply:
I still don't think having versions in there is a good idea. These things usually are quite dynamic. Imo better would be a generic text like "make sure to add the appropriate info.solidsoft.pitest plugin to your build.gradle or sth like that.

Reply:
You could do something like version "x.y.z" so the user understands, that he has to exchange it, I see the point where a user could think this is the latest version. But also I think it would not be overdramatic if it stays like this.

Reply:
Ok I changed it now ther version is x.y.z